cmd/skopeo/sync.go

```diff
@@ -723,6 +726,19 @@ func (opts *syncOptions) run(args []string, stdout io.Writer) (retErr error) {
 		logrus.Warn("Running in dry-run mode")
 	}
 
+	var digestFile *os.File
+	if opts.digestFile != "" && !opts.dryRun {
+		digestFile, err = os.OpenFile(opts.digestFile, os.O_TRUNC|os.O_CREATE|os.O_WRONLY, 0644)
+		if err != nil {
+			return fmt.Errorf("Error creating digest file: %w", err)
+		}
+		defer func() {
+			if errs := digestFile.Close(); errs != nil {
```

Comment:
If there were some other earlier error, that error is more likely to be a root cause, so it should be preferred.

… I forgot that we have a helper for this already: `noteCloseFailure`. Compare how it is used elsewhere.

Reply:
Fixed